websites/views.py

```diff
@@ -116,6 +117,11 @@ def get_queryset(self):
 
         if website_type is not None:
             queryset = queryset.filter(starter__slug=website_type)
+
+        if published is not None:
+            published = _parse_bool(published)
+            queryset = queryset.filter(publish_date__isnull=not published)
```

Comment:
So I have some concerns here about using `publish_date` as the mechanism for this functionality.  Currently when courses are imported, `publish_date` is brought in from the imported course.  This means that the value will reflect when the site was "published" according to the legacy Plone system.  We should either change this so that we import that as something like `legacy_publish_date` and ensure `publish_date` only refers to when it was published by `ocw-studio`, or use another property to check when the site was last published to Github.  I think in order to be referenced in a relation widget, we should also be checking this for both draft and live, so maybe we should `latest_build_id_draft` and `latest_build_id_live` and only display the result if both of those are not null?  What do you think, @mbertrand ?

Reply:
sure, that sounds straightforward enough

Reply:
Thinking about this some more, it might not be necessary to check `latest_build_id_draft` because `STATIC_API_BASE_URL` should be set to `https://ocwnext-rc.odl.mit.edu/` in RC and `https://ocwnext.odl.mit.edu` in Production, which would mean that both draft and live in RC for example would hit `https://ocwnext-rc.odl.mit.edu` with `getJSON`.

Reply:
I was mislead, btw, by our website factory, which if you ask for a published or unpublished website only varies the `publish_date` property, so I'll change that a bit to reflect this

Reply:
Are there any imported sites that have a `publish_date` but might not be actually published via concourse by the time this goes live?   If so, then it would make sense to add a separate `legacy_publish_date` for imported sites and upsert the `publish_date` field only when published via concourse.  If not, then I think it should be okay as is, though the additional field would provide clarification and might be worth doing anyway.  The `id` fields might not be a good indicator because those specific builds might have failed?

Reply:
@mbertrand That's a good point that I hadn't actually thought about, the builds could indeed have failed.  However, at least you know that if the site has been published at some point that the expected JSON file will exist.  When it comes to using that data in `ocw-hugo-themes`, it's up to the template itself to have logic to sanity check the data.  The main issue is when the file doesn't exist at all because Hugo throws a hard error if it gets anything other than a 200 when using `getJSON`.  I think the most ideal route here is to change `import_ocw_course_sites` to store the legacy publish date into a separate field and keep `publish_date` reserved for "when was the site last published live by `ocw-studio."  However, we would need to make that change likely in a separate PR first, merge it before this one and make sure all courses are re-imported before this is merged.  Said PR would also need to include a migration to make sure `publish_date` is null where it's supposed to be for imported courses that haven't been published yet.

Reply:
am I understanding correctly that based on the data we currently store in the database there is not a foolproof way to determine whether a course has already been published through concourse?

I think we could determine the following things, correct me if I'm wrong:

1. If there is no `latest_build_id_live` (or `..._draft`) then the site has _not_ been published through concourse
2. if there is a value for `latest_build_id_{draft|live}`, then:
    1. if `{live|draft}_publish_status` is `PUBLISH_STATUS_SUCCEEDED`, then the course was published! 🎉 
    2. if `{live|draft}_publish_status` is _not_ `PUBLISH_STATUS_SUCCEEDED`, then we know that the _most recent_ build failed, aborted, is pending, etc. But just looking around a little bit it seems like we don't have a way of knowing whether a _previous_ build has ever succeeded - another way to say that is that I don't see anywhere where we're storing the build history. If we were then I think we could easily just check to see if there has ever been a successful build of the site.

I'm wondering in particular about the 2-ii case - is it true that we can't determine the status of previous attempts to publish this course?

if that's correct, then I think the suggestion to add `legacy_publish_date` makes sense, so that then we can have a property (`publish_date`) which serves as an unambiguous record to indicate that _yes, this has been published_.

About the legacy data - is there anything we need it for? If not, would it also be possible to just drop that info on the floor? the publish date on the plone CMS may not be important, I don't know.

In any case, my assumption reading the code and not knowing about the whole publish pipeline was that I'd be able to rely on a property like `publish_date` to tell me when something had been published through our system. If that value doesn't represent that, then I think we should refactor it so that it does store that.

Reply:
@pdpinch What do you think about this?  Should we be importing `publish_date` from the Plone data at all?  Should we preserve it as `legacy_publish_date`?

Reply:
I spoke with @pdpinch about this offline and here's the plan:

1. Alter `import_ocw_course_sites` to store `publishdate` from the `ocw-to-hugo` data as a new field called `first_published_to_production` (this is what it's called in the original source data) instead of in `publish_date`
2. On first publish of a course in `ocw-studio`, set both `first_published_to_production` and `publish_date`, then on subsequent publishes only set `publish_date`

Reply:
I made an issue about this here: https://github.com/mitodl/ocw-studio/issues/916

Once it's addressed, we should be able to simply merge this as-is.  Until then, I think we can consider this blocked.